website/src/supportedNetworks.tsx

```diff
@@ -34,33 +89,323 @@ export async function getSupportedNetworksStaticProps() {
+      )
+    }
+
+    return filtered
+  }, [networks, searchQuery, showTestnets])
+
+  useEffect(() => {
+    setNoResults(filteredNetworks.length === 0 && searchQuery !== '')
+  }, [filteredNetworks.length, searchQuery])
+
+  const handleClearSearch = () => {
+    setSearchQuery('')
+  }
+
+  const handleShowTestnets = () => {
+    if (noResults) {
+      setSearchQuery('')
+    }
+
+    setIsLoadingTestnets(true)
+
+    setTimeout(() => {
+      setShowTestnets(true)
+      setIsLoadingTestnets(false)
+    }, 150)
+  }
+
+  const handleHideTestnets = () => {
+    setShowTestnets(false)
+  }
+
+  const handleToggleTestnets = () => {
+    if (!showTestnets) {
+      handleShowTestnets()
+    } else {
+      handleHideTestnets()
+    }
+  }
+
+  // Generate skeleton rows
+  const skeletonRows = Array.from({ length: 20 }, (_, index) => <SkeletonRow key={`skeleton-${index}`} />)
+
+  const shouldShowLoadingScreen = isLoading || (isLoadingTestnets && !hasLoadedDataBefore)
 
   return (
-    <Table>
-      <tbody>
-        <tr>
-          <th>{t('supportedNetworks.name')}</th>
-          <th>{t('supportedNetworks.id')}</th>
-          <th align="center">{t('supportedNetworks.subgraphs')}</th>
-          <th align="center">{t('supportedNetworks.substreams')}</th>
-          <th align="center">{t('supportedNetworks.firehose')}</th>
-        </tr>
-        {networks.map((network) => (
-          <tr key={network.id}>
-            <td>{network.fullName}</td>
-            <td>
-              <code className="font-mono">{network.id}</code>
-            </td>
-            <td align="center">{network.subgraphs ? '✓' : null}</td>
-            <td align="center">{network.substreams ? '✓' : null}</td>
-            <td align="center">{network.firehose ? '✓' : null}</td>
-          </tr>
-        ))}
-      </tbody>
-    </Table>
+    <>
+      <Callout variant="info" className="mb-6 flex items-center">
+        <Text.P14>
+          {t('index.supportedNetworks.infoText')}{' '}
+          <ExperimentalLink href="https://thegraph.com/docs/en/indexing/chain-integration-overview/">
+            {t('index.supportedNetworks.infoLink')}
+          </ExperimentalLink>
+          .
+        </Text.P14>
+      </Callout>
+
+      <div className="mb-4 flex items-center gap-4">
+        <div className="flex-grow">
+          <ExperimentalSearch
+            placeholder={t('index.supportedNetworks.search')}
+            size="small"
+            value={searchQuery}
+            onChange={(value) => setSearchQuery(value)}
+            className="w-full"
+          />
+        </div>
+        <ExperimentalToggleChip
+          size="small"
+          onChange={handleToggleTestnets}
+          checked={showTestnets}
+          disabled={isLoadingTestnets}
+          iconBefore={isLoadingTestnets ? <LoadingSpinner className="size-4" /> : undefined}
+        >
+          {t('index.supportedNetworks.showTestnets')}
+        </ExperimentalToggleChip>
+      </div>
```

Comment:
```suggestion
        <ExperimentalToggleChip
          size="small"
          onChange={handleToggleTestnets}
          checked={showTestnets}
        >
          {t('index.supportedNetworks.showTestnets')}
        </ExperimentalToggleChip>
      </div>
```

One of the biggest design decisions that we made while designing these new Chip components was to ensure that their length always stays the same, no layout shifts, no other content pushes. we tried very hard to achieve this. And now, I see with this approach, the Search bar is being pushed because of the loading state. 

I seriously don't believe we need a loading state, this is almost instant (and there is no server/api calls to filter out the list) so I tried in my local how it feels without this loading state, and i feels good. Can we please remove the loading states here? 

https://share.cleanshot.com/Jt0mPrcT